extension/encoding/googlecloudlogentryencodingextension/internal/vpcflowlog/parser.go

```diff
@@ -0,0 +1,327 @@
+	if ns == nil {
+		return
+	}
+
+	shared.PutInt(gcpVPCFlowNetworkServiceDSCP, ns.DSCP, attr)
+}
+
+func handleInstance(inst *instance, side flowSide, attr pcommon.Map) error {
+	if inst == nil {
+		return nil
+	}
+
+	// Validate that side is one of our constants
+	if side != src && side != dest {
+		return errors.New("unsupported side passed to handleInstance. Must be one of src or dest")
+	}
+
+	shared.PutStr(fmtAttributeNameUsingSide(gcpVPCFlowInstanceProjectIDTemplate, side), inst.ProjectID, attr)
+	shared.PutStr(fmtAttributeNameUsingSide(gcpVPCFlowInstanceVMRegionTemplate, side), inst.Region, attr)
+	shared.PutStr(fmtAttributeNameUsingSide(gcpVPCFlowInstanceVMNameTemplate, side), inst.VMName, attr)
+	shared.PutStr(fmtAttributeNameUsingSide(gcpVPCFlowInstanceVMZoneTemplate, side), inst.Zone, attr)
+
+	if inst.ManagedInstanceGroup != nil {
+		shared.PutStr(fmtAttributeNameUsingSide(gcpVPCFlowInstanceMIGNameTemplate, side), inst.ManagedInstanceGroup.Name, attr)
+		shared.PutStr(fmtAttributeNameUsingSide(gcpVPCFlowInstanceMIGZoneTemplate, side), inst.ManagedInstanceGroup.Zone, attr)
+	}
+
+	return nil
+}
+
+func handleLocation(loc *location, side flowSide, attr pcommon.Map) error {
+	if loc == nil {
+		return nil
+	}
+
+	// Validate that side is one of our constants
+	if side != src && side != dest {
+		return errors.New("unsupported side passed to handleLocation. Must be one of src or dest")
+	}
+
+	shared.PutInt(fmtAttributeNameUsingSide(gcpVPCFlowASNTemplate, side), loc.ASN, attr)
+	shared.PutStr(fmtAttributeNameUsingSide(gcpVPCFlowGeoCityTemplate, side), loc.City, attr)
+	shared.PutStr(fmtAttributeNameUsingSide(gcpVPCFlowGeoContinentTemplate, side), loc.Continent, attr)
+	shared.PutStr(fmtAttributeNameUsingSide(gcpVPCFlowGeoCountryTemplate, side), loc.Country, attr)
+	shared.PutStr(fmtAttributeNameUsingSide(gcpVPCFlowGeoRegionTemplate, side), loc.Region, attr)
+
+	return nil
+}
+
+func handleVPC(vpc *vpc, side flowSide, attr pcommon.Map) error {
+	if vpc == nil {
+		return nil
+	}
+
+	// Validate that side is one of our constants
+	if side != src && side != dest {
+		return errors.New("unsupported side passed to handleVPC. Must be one of src or dest")
+	}
+
+	shared.PutStr(fmtAttributeNameUsingSide(gcpVPCFlowProjectIDTemplate, side), vpc.ProjectID, attr)
+	shared.PutStr(fmtAttributeNameUsingSide(gcpVPCFlowSubnetNameTemplate, side), vpc.SubnetworkName, attr)
+	shared.PutStr(fmtAttributeNameUsingSide(gcpVPCFlowSubnetRegionTemplate, side), vpc.SubnetworkRegion, attr)
+	shared.PutStr(fmtAttributeNameUsingSide(gcpVPCFlowVPCNameTemplate, side), vpc.VPCName, attr)
+
+	return nil
+}
+
+func handleInternetRoutingDetails(ird *internetRoutingDetails, attr pcommon.Map) {
+	if ird == nil || len(ird.EgressASPath) == 0 {
+		return
+	}
+
+	asPaths := attr.PutEmptySlice(gcpVPCFlowEgressASPaths)
+	for _, path := range ird.EgressASPath {
+		pathMap := asPaths.AppendEmpty().SetEmptyMap()
+		asDetails := pathMap.PutEmptySlice("as_details")
```

Comment:
Give that `asDetails` is a struct and here is an empty slice, let's not reuse this name for this use case

<img width="741" height="143" alt="image" src="https://github.com/user-attachments/assets/f208befa-b158-448e-872c-66161ccb1431" />